python/src/uagents/agent.py

```diff
@@ -508,6 +513,36 @@ def initialize_wallet_messaging(
         else:
             self._wallet_messaging_client = None
 
+    def _initialize_metadata(self, metadata: Optional[Dict[str, Any]]):
+        """
+        Initialize the metadata for the agent.
+
+        The metadata is filtered to include only location-based metadata and the
+        model ensures that the metadata is valid and complete.
+
+        Args:
+            metadata (Dict[str, Any]): The metadata to include in the agent object.
+        """
+        if not metadata or "geolocation" not in metadata:
+            return {}
+
+        class AgentGeolocation(BaseModel):
```

Comment:
We will definitely use this set (lat, lng, radius) more often at different places in the Agent. I like the idea of making it available in the `types.py`